ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionLogUpdatesConversions.scala

```diff
@@ -44,13 +45,16 @@ private[events] object TransactionLogUpdatesConversions {
         }
         val filteredFlatEvents = flatTransactionEvents
           .filter(flatTransactionPredicate(wildcardParties, templateSpecificParties))
-        val hasOneEventWithCommandId = filteredFlatEvents.exists(_.commandId.nonEmpty)
+        val commandId = getCommandId(filteredFlatEvents, requestingParties)
+        val nonTransient = removeTransient(filteredFlatEvents)
```

Comment:
But now we compute removeTransient twice, no?

Reply:
Actually I wanted to be sure not to break anything. Is `toApiTransaction` **only** called after `filter`? I could not figure out.

Reply:
Yes, they are called one after another but kept as separate units for performance considerations.